cli/keys.go

```diff
@@ -3,66 +3,125 @@
 package cli
 
 import (
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
 	"crypto/rand"
 	"crypto/rsa"
 	"crypto/x509"
 	"encoding/pem"
 	"log"
 	"os"
+	"reflect"
 
 	"github.com/spf13/cobra"
 )
 
 const (
 	keyBitSize     = 4096
-	privateKeyType = "RSA PRIVATE KEY"
+	rsaKeyType     = "RSA PRIVATE KEY"
```

Comment:
please use openssl standards for encoding the keys, you can use this commands to compare, since a user might generate their keys externally not using cli

```shell
openssl genrsa -out rsa_private.key 2048
openssl rsa -in rsa_private.key -pubout -out rsa_public.key
openssl ecparam -name prime256v1 -genkey -noout -out ecdsa_private.key
openssl ec -in ecdsa_private.key -pubout -out ecdsa_public.key
openssl genpkey -algorithm ED25519 -out ed25519_private.key
openssl pkey -in ed25519_private.key -pubout -out ed25519_public.key
```